File: 프로그래머스/2/42584. 주식가격/주식가격.py
def solution(prices):
    answer = []
    stack = []
    temp = []
    
    for index, i in enumerate(prices) :
        
        while stack and stack[-1][-1] > i :
            t = stack.pop()
            temp.append((t[0], index - t[0]))
        stack.append((index, i))
    
    else :
        while stack :
            t = stack.pop()
            temp.append((t[0], index - t[0]))
        
    temp.sort(key = lambda x : x[0])
    for i in temp :
        answer.append(i[1])
        
    return answer

File: 프로그래머스/2/42584. 주식가격/주식가격.java
import java.util.*;

class Solution {
public int[] solution(int[] prices) {
        int[] answer = new int[prices.length];
        int last_index = prices.length - 1;
        ArrayDeque<Integer[]> stack = new ArrayDeque<>();
        
        for (int i = 0; i < prices.length; i++) {
            while (!stack.isEmpty() && stack.peekLast()[1] > prices[i]) {
                Integer[] pop = stack.pollLast();
                answer[pop[0]] = i - pop[0];
            }
            stack.addLast(new Integer[]{i, prices[i]});
        }

        while (!stack.isEmpty()) {
            Integer[] pop = stack.pollLast();
            answer[pop[0]] = last_index - pop[0];
        }

        return answer;
    }
}